# HG changeset patch
# User Lance Albertson <lance@osuosl.org>
# Date 1256163765 25200
# Node ID 082287c0fdc321de66cda28a422c8faab99708c8
# Parent  9e649337acf061d12439034ca51148929dccdb58
Use linux32/linux64 to properly set the arch for the machine

diff -r 9e649337acf0 -r 082287c0fdc3 bin/rinse
--- a/bin/rinse	Tue Apr 21 19:29:19 2009 +0100
+++ b/bin/rinse	Wed Oct 21 15:22:45 2009 -0700
@@ -204,6 +204,7 @@
 $CONFIG{ 'cache-dir' } = "/var/cache/rinse/";
 $CONFIG{ 'config' }    = "/etc/rinse/rinse.conf";
 $CONFIG{ 'pkgs-dir' }  = "/etc/rinse";
+$CONFIG{ 'setarch' }   = "linux32";
 
 
 #
@@ -226,6 +227,15 @@
 
 
 #
+# Set the proper setarch command depending on the arch
+#
+if ( $CONFIG{ 'arch' } eq 'amd64' )
+{
+    $CONFIG{ 'setarch' } = 'linux64';
+}
+
+
+#
 #  Handle special case first.
 #
 if ( $CONFIG{ 'list-distributions' } )
@@ -1121,7 +1131,7 @@
         {
             my $rpmname = basename($file);
             $postcmd =
-              "cp $file $CONFIG{'directory'}/tmp ; chroot $CONFIG{'directory'} rpm -ivh --force --nodeps /tmp/$rpmname ; rm $CONFIG{'directory'}/tmp/$rpmname";
+              "cp $file $CONFIG{'directory'}/tmp ; $CONFIG{'setarch'} chroot $CONFIG{'directory'} rpm -ivh --force --nodeps /tmp/$rpmname ; rm $CONFIG{'directory'}/tmp/$rpmname";
         }
         system($cmd );
 
@@ -1185,7 +1195,7 @@
     {
         print "Running custom script: $CONFIG{'before-post-install'}\n";
 
-        system( $CONFIG{ 'before-post-install' }, $prefix );
+        system("$CONFIG{ 'setarch' } $CONFIG{ 'before-post-install' }", $prefix );
     }
 
     #
@@ -1196,7 +1206,7 @@
     {
         print "Running custom script: $CONFIG{'post-install'}\n";
 
-        system( $CONFIG{ 'post-install' }, $prefix );
+        system("$CONFIG{ 'setarch' } $CONFIG{ 'post-install' }", $prefix );
         return;
     }
 
@@ -1220,7 +1230,7 @@
         }
         print "Running post-install script $name:\n";
         my $cmd = "$file $prefix";
-        system($cmd );
+        system("$CONFIG{ 'setarch' } $cmd" );
 
         $CONFIG{ 'verbose' } && print "+script $file\n";
     }
@@ -1233,7 +1243,7 @@
     {
         print "Running custom script: $CONFIG{'after-post-install'}\n";
 
-        system( $CONFIG{ 'after-post-install' }, $prefix );
+        system("$CONFIG{ 'setarch' } $CONFIG{ 'after-post-install' }", $prefix );
     }
 
 }
